### PR TITLE
Metadata source may differ from circulation source

### PR DESCRIPTION
**Pull request by leonardr**
This branch reintroduces the idea that, when importing data from an OPDS feed, the actual book might come from a different source than the OPDS feed, and that the <bibframe:distribution> tag says where the actual book comes from.

Currently this only happens on the open-access content server. The Metadata created as part of the import is associated with the source of the OPDS feed (this includes any image links), but the CirculationData is associated with the source of the book. This means that you can end up with a Work whose LicensePool comes from Gutenberg, but the presentation Edition of that LicensePool comes from the open-access content server.

The motivating case here is that if a circ manager finds six editions of a book from the open-access content server, and every one of them has the same data source ("content server"), the circ manager has no way of knowing which edition it should use. When the data sources get passed through, the circ manager can make a rough judgement that certain data sources produce higher-quality books than others.

### CODE REVIEW

File: opds_import.py [hunk 1]
```diff
@@ -556,7 +552,40 @@ def _data_detail_for_feedparser_entry(cls, entry, data_source):
...
+                _db = Session.object_session(metadata_data_source)
+                circulation_data_source = DataSource.lookup(
+                    _db, circulation_data_source_name
+                )
+                if not circulation_data_source:
+                    raise ValueError(
+                        "Unrecognized circulation data source: %s" % (
+                            circulation_data_source_name
+                        )
+                    )
```
Comment (leonardr) [inline opds_import.py]: If a data source is added to the content server, but the circ manager doesn't know about it yet, this code stops a crash.

File: opds_import.py [hunk 2]
```diff
@@ -607,10 +636,12 @@ def summary_to_linkdata(detail):
             data_source_last_updated=last_opds_update,
         )
             
-        # Only add circulation data if the data source is lendable.
-        if data_source.offers_licenses:
+        # Only add circulation data if both the book's distributor *and*
+        # the source of the OPDS feed are lendable data sources.
+        if (circulation_data_source and circulation_data_source.offers_licenses
+            and metadata_data_source.offers_licenses):
```
Comment (leonardr) [inline opds_import.py]: This means that even if the metadata wrangler purports to offer licenses to books, we won't take it up on the offer.

I think this only happens in the tests, but it's good to fix the tests in a way that will stop a bad thing from happening in real life.

Comment (aslagle) [inline opds_import.py]: Should we change the content server data source back to not offering licenses? And then we wouldn't need to check the metadata_data_source here.

**Comment by daryanypl** [comment]
👍

File: opds_import.py (hunk 2 above)
Comment (leonardr) [inline opds_import.py]: I don't think we really need it either, it's just a safety measure.

**Comment by aslagle** [comment]
:+1:

**Pull request merged**